src/modules/cpu/kernel/glitch.hpp

```diff
@@ -0,0 +1,1133 @@
+        dstPtrRowG += dstDescPtr->strides.hStride;
+        dstPtrRowB += dstDescPtr->strides.hStride;
+    }
+}
+
+RppStatus glitch_u8_u8_host_tensor(Rpp8u *srcPtr,
+                                   RpptDescPtr srcDescPtr,
+                                   Rpp8u *dstPtr,
+                                   RpptDescPtr dstDescPtr,
+                                   Rpp32u *xOffsetR,
+                                   Rpp32u *yOffsetR,
+                                   Rpp32u *xOffsetG,
+                                   Rpp32u *yOffsetG,
+                                   Rpp32u *xOffsetB,
+                                   Rpp32u *yOffsetB,
+                                   RpptROIPtr roiTensorPtrSrc,
+                                   RpptRoiType roiType,
+                                   RppLayoutParams layoutParams)
+{
+    RpptROI roiDefault = {0, 0, (Rpp32s)srcDescPtr->w, (Rpp32s)srcDescPtr->h};
+    omp_set_dynamic(0);
+#pragma omp parallel for num_threads(dstDescPtr->n)
+    for (int batchCount = 0; batchCount < dstDescPtr->n; batchCount++)
+    {
+        __m128i maskR = _mm_setr_epi8(0, 0x80, 0x80, 3, 0x80, 0x80, 6, 0x80, 0x80, 9, 0x80, 0x80, 12, 0x80, 0x80, 15);
+        __m128i maskGB = _mm_setr_epi8(0x80, 1, 2, 0x80, 4, 5, 0x80, 7, 8, 0x80, 10, 11, 0x80, 13, 14, 0x80);
+        RpptROI roi;
+        RpptROIPtr roiPtrInput = &roiTensorPtrSrc[batchCount];
+        compute_roi_validation_host(roiPtrInput, &roi, &roiDefault, roiType);
+
+        Rpp32u xOffsetRchn = xOffsetR[batchCount];
+        Rpp32u yOffsetRchn = yOffsetR[batchCount];
+        Rpp32u xOffsetGchn = xOffsetG[batchCount];
+        Rpp32u yOffsetGchn = yOffsetG[batchCount];
+        Rpp32u xOffsetBchn = xOffsetB[batchCount];
+        Rpp32u yOffsetBchn = yOffsetB[batchCount];
+
+        Rpp32u elementsInRowMax = srcDescPtr->w;
+
+        Rpp32u xOffsets[3] = {xOffsetRchn, xOffsetGchn, xOffsetBchn};
+        Rpp32u yOffsets[3] = {yOffsetRchn, yOffsetGchn, yOffsetBchn};
+        Rpp32u xOffsetsLoc[3] = {xOffsetRchn, xOffsetGchn, xOffsetBchn};
+        Rpp32u yOffsetsLoc[3] = {yOffsetRchn * elementsInRowMax, yOffsetGchn * elementsInRowMax, yOffsetBchn * elementsInRowMax};
+
+        Rpp8u *srcPtrImage, *dstPtrImage;
+        srcPtrImage = srcPtr + batchCount * srcDescPtr->strides.nStride;
+        dstPtrImage = dstPtr + batchCount * dstDescPtr->strides.nStride;
+
+        Rpp32u bufferLength = roi.xywhROI.roiWidth * layoutParams.bufferMultiplier;
+        Rpp8u *srcPtrChannel, *dstPtrChannel;
+        srcPtrChannel = srcPtrImage + (roi.xywhROI.xy.y * srcDescPtr->strides.hStride) + (roi.xywhROI.xy.x * layoutParams.bufferMultiplier);
+        dstPtrChannel = dstPtrImage;
+
+        if ((srcDescPtr->c == 3) && (srcDescPtr->layout == RpptLayout::NHWC) && (dstDescPtr->layout == RpptLayout::NCHW))
+        {
+            convert_pkd3_pln3<Rpp8u>(srcPtrImage, dstPtrImage, srcDescPtr, dstDescPtr, roi.xywhROI.roiWidth * 3);
```

Comment:
Why aren't we using the internal shuffled loads and instead doing a preconvert here?